source/extensions/stat_sinks/metrics_service/grpc_metrics_service_impl.cc

```diff
@@ -38,20 +38,17 @@ void GrpcMetricsStreamerImpl::send(envoy::service::metrics::v3::StreamMetricsMes
 }
 
 MetricsServiceSink::MetricsServiceSink(const GrpcMetricsStreamerSharedPtr& grpc_metrics_streamer,
-                                       TimeSource& time_source,
                                        const bool report_counters_as_deltas)
-    : grpc_metrics_streamer_(grpc_metrics_streamer), time_source_(time_source),
+    : grpc_metrics_streamer_(grpc_metrics_streamer),
       report_counters_as_deltas_(report_counters_as_deltas) {}
 
 void MetricsServiceSink::flushCounter(
-    const Stats::MetricSnapshot::CounterSnapshot& counter_snapshot) {
+    const Stats::MetricSnapshot::CounterSnapshot& counter_snapshot, int64_t snapshot_time) {
```

Comment:
either pass a SystemTime here or include the units in the var name (snapshot_time_ms). Mostly I think we should pass around time objects rather than pre-converted ones, but I guess doing it this way factors out a conversion that would otherwise be repeated many times. If that's the reason, can you put that in a comment?

Reply:
Yes.  That is the reason. I will add a comment where the variable is declared.